gdal/frmts/gtiff/libtiff/tif_webp.c

```diff
@@ -143,28 +143,27 @@ TWebPDecode(TIFF* tif, uint8* op, tmsize_t occ, uint16 s)
     }
     return 0;
   } else {
-    int current_y, width, stride;
+    int current_y, stride;
     uint8_t* buf;
-    
+
     // Returns the RGB/A image decoded so far
-    buf = WebPIDecGetRGB(sp->psDecoder, &current_y, &width, NULL, &stride);
-    
-    if ((current_y - sp->last_y) * width * stride < occ) {
+    buf = WebPIDecGetRGB(sp->psDecoder, &current_y, NULL, NULL, &stride);
```

Comment:
shouldn't buf be test against NULL ? for example in case of a corrupted/truncated stream